tests/test_location_pi.py

```diff
@@ -0,0 +1,508 @@
+        """
+        Successfully creates the expected picking when a quant is
+        specified.
+        """
+        quant = self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                                  package_id=self.package_one.id)
+        count_moves = [{"quant_ids": [quant.id],
+                        "location_id": self.test_location_01,
+                        "location_dest_id": self.test_location_02.id}]
+
+        pickings = self.test_location_01._process_pi_count_moves(count_moves)
+
+        self.assertEqual(len(pickings), 1,
+                         "Did not create exactly one picking")
+        self._check_picking_loc01_to_loc02(pickings[0], 0)
+        self.assertEqual(pickings[0].move_line_ids[0].package_id.id,
+                         self.package_one.id,
+                         "Picking is not related to the specified package")
+
+    def test10_process_pi_count_moves_bad_request_with_reserved_package(self):
+        """
+        Throws a ValidationError in case the specified package is
+        reserved. Checking by specifying both package / quants list.
+        """
+        quant = self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                                  package_id=self.package_one.id)
+        self.create_picking(self.picking_type_pick,
+                            products_info=[{'product': self.apple,
+                                            'qty': 4}],
+                            confirm=True,
+                            assign=True)
+        requests = [
+            [{
+                "package_id": self.package_one.id,
+                "location_id": self.test_location_01,
+                "location_dest_id": self.test_location_02.id
+            }],
+            [{
+                "quant_ids": [quant.id],
+                "location_id": self.test_location_01,
+                "location_dest_id": self.test_location_02.id
+            }]
+        ]
+
+        for count_moves in requests:
+            with self.assertRaisesRegex(ValidationError,
+                                        "Some quants are already reserved"):
+                self.test_location_01._process_pi_count_moves(count_moves)
+
+    #
+    ## PI Inventory Adjustments
+    #
+
+    def test11__process_inventory_adjustments_returns_None_when_no_drift(self):
+        """ Returns None when no request item is specified """
+        inv = self.test_location_01._process_inventory_adjustments([])
+
+        self.assertIsNone(inv, "Returned a new inventory for empty request")
+
+    def test12_get_stock_drift_no_package_token(self):
+        """
+        Returns the correct metadata for the picking creation if
+        a request item contains a 'NO_PACKAGE' package.
+        """
+        self.create_quant(self.apple.id, self.test_location_01.id, 4)
+        adjs_req = [{"product_id": self.apple.id,
+                     "package_name": "tes12_NO_PACKAGE",
```

Comment:
`tes12` is this a typo or inteneded? same happens later in tes14.

Reply:
In fact, NO_PACKAGE alone is what we should receive.

Reply:
where's the typo?

Reply:
ah, `tes`, it's just a typo, but it doesn't matter ( you made me recheck the token in the old code...)